x-pack/plugin/ent-search/src/main/java/org/elasticsearch/xpack/application/search/SearchApplication.java

```diff
@@ -175,19 +177,42 @@ public static SearchApplication fromXContent(String resourceName, XContentParser
     }
 
     /**
-     * Converts the {@link SearchApplication} to XContent.
+     * Adds base fields to {@link XContentBuilder} object.
      *
-     * @return The {@link XContentBuilder} containing the serialized {@link SearchApplication}.
      */
-    @Override
-    public XContentBuilder toXContent(XContentBuilder builder, Params params) throws IOException {
-        builder.startObject();
+    private void buildBaseXContent(XContentBuilder builder, boolean includeIndex) throws IOException {
         builder.field(NAME_FIELD.getPreferredName(), name);
+        if (includeIndex) {
+            builder.field(INDICES_FIELD.getPreferredName(), indices);
+        }
         if (analyticsCollectionName != null) {
             builder.field(ANALYTICS_COLLECTION_NAME_FIELD.getPreferredName(), analyticsCollectionName);
         }
         builder.field(UPDATED_AT_MILLIS_FIELD.getPreferredName(), updatedAtMillis);
         builder.field(TEMPLATE_FIELD.getPreferredName(), searchApplicationTemplate);
+    }
+
+    /**
+     * Converts the {@link SearchApplication} to XContent.
+     *
+     * @return The {@link XContentBuilder} containing the serialized {@link SearchApplication}.
+     */
+    @Override
+    public XContentBuilder toXContent(XContentBuilder builder, Params params) throws IOException {
+        builder.startObject();
+        buildBaseXContent(builder, false);
+        builder.endObject();
+        return builder;
+    }
+
+    /**
+     * Converts the {@link SearchApplication} to XContent.
+     *
+     * @return The {@link XContentBuilder} containing the serialized {@link SearchApplication}.
+     */
+    public XContentBuilder toXContentWithIndex(XContentBuilder builder) throws IOException {
```

Comment:
I'm not thrilled with this because we now have a production `toXContent` that's not tested with the rest of our flow. I think this might evolve though with other PR feedback.

Reply:
Any thoughts on implement this part in other way? 🤔

Reply:
If indices are null, we don't write them. Would that be an option?